openstudiocore/CMakeLists.txt

```diff
@@ -551,46 +561,32 @@ if(BUILD_WITH_OPENSSL)
   endif()
 endif()
 
-# swig
-
-set(SWIG_INSTALL_DIR "" CACHE PATH "Path to swig Install")
-mark_as_advanced(SWIG_INSTALL_DIR)
-
-if(NOT EXISTS ${SWIG_INSTALL_DIR})
-  if( WIN32 )
-    set(SWIG_ZIP_FILENAME "swig_3_0_10_Win32.zip")
-    set(SWIG_ZIP_EXPECTED_MD5 "be3c42e6daffde06a05ad1b978338ebd")
-  elseif( APPLE )
-    set(SWIG_ZIP_FILENAME "swig_3_0_10_osx.tar.gz")
-    set(SWIG_ZIP_EXPECTED_MD5 "f0ea8749f8a0345ee37e1766d27a984d")
-  elseif(EXISTS "/etc/redhat-release")
-  else()
-  endif()
-
-  set(SWIG_ZIP_LOCAL_PATH "${CMAKE_BINARY_DIR}/${SWIG_ZIP_FILENAME}")
-  if(EXISTS "${SWIG_ZIP_LOCAL_PATH}")
-    file(MD5 "${SWIG_ZIP_LOCAL_PATH}" SWIG_ZIP_MD5)
-  endif()
+# SWIG
 
-  if( NOT SWIG_ZIP_MD5 STREQUAL SWIG_ZIP_EXPECTED_MD5)
-    file(DOWNLOAD "http://openstudio-resources.s3.amazonaws.com/dependencies/${SWIG_ZIP_FILENAME}"
-      ${SWIG_ZIP_LOCAL_PATH}
-      INACTIVITY_TIMEOUT 120
-      SHOW_PROGRESS
-      EXPECTED_MD5 ${SWIG_ZIP_EXPECTED_MD5})
-      execute_process(COMMAND ${CMAKE_COMMAND} -E tar xfz ${SWIG_ZIP_LOCAL_PATH} WORKING_DIRECTORY "${CMAKE_BINARY_DIR}")
-  endif()
-
-  set(SWIG_INSTALL_DIR "${CMAKE_BINARY_DIR}/Swig-install/")
+if(UNIX)
+  # We patch it up with a version of pcre we provide to avoid having to have the requirement locally
+  ExternalProject_Add(SWIG
```

Comment:
I think this is a mistake to use external project. (Now one more level deep in the inception of projects) How about just a custom build rule? You are already predicting where the swig executable is on line 577. I think the reason to external project would be to have the opportunity to cmake configure to whatever the external project products where. Here you're not doing that so why add such complexity? Perhaps there is no real tangible bad side effect of this, but when i was building this the other day my configure failed where you where trying to you use "SWIG" as a target level dependency on another target. I just seems bad. Just add_custom_command with output being the swig executable and use that as a dependency on the targets that need swig.

Or better yet. just use the silly swig lib environment variable or cli argument or whatever.

I know I introduced EP. But I regret it. Still has a place. like where we are still using it to create the binary packs of dependencies we upload to s3. But it has caused pain, and now I'm afraid of it. Here I particularly don't see the value.